image/util.go

```diff
@@ -55,14 +55,21 @@ func GetClusterFileFromImage(imageName string) string {
 // GetClusterFileFromImageManifest retrieve ClusterFile from image manifest(image yaml)
 func GetClusterFileFromImageManifest(imageName string) string {
 	//  find cluster file from image manifest
-	imageMetadata, err := NewImageMetadataService().GetRemoteImage(imageName)
+	var image *v1.Image
+	var err error
+	image, err = imageUtils.GetImage(imageName)
 	if err != nil {
-		return ""
+		imageMetadata, err := NewImageMetadataService().GetRemoteImage(imageName)
+		if err != nil {
+			logger.Error("failed to found image %s,err: %v", imageName, err)
+			return ""
+		}
+		image = &imageMetadata
 	}
-	if imageMetadata.Annotations == nil {
+	if image.Annotations == nil {
```

Comment:
this line is redudant. you can remove it.

When a map is nil or length of map is zero, you can always use `value, ok := image.Annotations[key]` to check if your key exists.